tests/test_quixstreams/test_dataframe/test_dataframe.py

```diff
@@ -541,6 +543,30 @@ def test_drop_missing_columns_errors_ignore(self, dataframe_factory):
             0
         ] == (expected, key, timestamp, headers)
 
+    def test__dataframe_clone__stream_id(
+        self, dataframe_factory, topic_manager_topic_factory
+    ):
+        sdf = dataframe_factory(stream_id="some-id")
+        assert sdf.stream_id == "some-id"
+
+        # Ensure that the previous stream_id is passed
+        assert sdf.__dataframe_clone__().stream_id == sdf.stream_id
+
+        # Ensure that new stream_id is passed
+        assert sdf.__dataframe_clone__(stream_id="new-id").stream_id == "new-id"
+
+        # If the topic is passed to the __dataframe_clone__, the existing
+        # must be overridden
```

Comment:
I think you missed this
```suggestion
        # stream_id must be overridden
```